ssg/playbook_builder.py

```diff
@@ -0,0 +1,275 @@
+            if isinstance(item, str):
+                match = xccdf_var_pattern.match(item)
+                if match:
+                    var_id = match.group(1)
+                    value = self.choose_variable_value(var_id, variables,
+                                                       refinements)
+                    values[var_id] = value
+                else:
+                    raise ValueError("Found unknown item '%s'" % item)
+            else:
+                tasks.append(item)
+        return tasks, values
+
+    def get_tags_from_comments(self, snippet_path):
+        """
+        Adds tags based on recognized comments in the given Ansible snippet.
+        """
+        tags = []
+        remediation = ssg.build_remediations.parse_from_file_without_jinja(
+            snippet_path
+        )
+        config = remediation.config
+        for k, v in config.items():
+            if k in COMMENTS_TO_PARSE:
+                tags.append("%s_%s" % (v, k))
+        return tags
+
+    def get_benchmark_variables(self):
+        """
+        Get all variables, their selectors and values used in a given
+        benchmark. Returns a dictionary where keys are variable IDs and
+        values are dictionaries where keys are selectors and values are
+        variable values.
+        """
+        variables = dict()
+        for dirpath, dirnames, filenames in os.walk(self.guide_dir):
+            for filename in filenames:
+                root, ext = os.path.splitext(filename)
+                if ext == ".var":
+                    full_path = os.path.join(dirpath, filename)
+                    xccdf_value = ssg.build_yaml.Value.from_yaml(full_path)
+                    # Make sure that selectors and values are strings
+                    options = dict()
+                    for k, v in xccdf_value.options.items():
+                        options[str(k)] = str(v)
+                    variables[xccdf_value.id_] = options
+        return variables
+
+    def create_playbook(self, snippet_path, rule_id, variables,
+                        refinements, output_dir):
+        """
+        Creates a Playbook from Ansible snippet for the given rule specified
+        by rule ID, fills in the profile values and saves it into output_dir.
+        """
+
+        with open(snippet_path, "r") as snippet_file:
+            snippet_str = snippet_file.read()
+        snippet_yaml = ssg.yaml.ordered_load(snippet_str)
+
+        play_tasks, play_vars = self.get_data_from_snippet(
+            snippet_yaml, variables, refinements
+        )
+
+        if len(play_tasks) == 0:
+            raise ValueError(
+                "Ansible remediation for rule '%s' in '%s' "
+                "doesn't contain any task." %
+                (rule_id, snippet_path)
+            )
+
+        tags = set()
+        for task in play_tasks:
+            tags |= set(task.pop("tags", []))
+
+        # play should be a dictionary, but to keep a logical order of elements
```

Comment:
If keys are added to `OrderedDict` one-by-one, there is no need for the sequence, or for the comment. Although your approach is correct, I would favour explicit assignment.

Reply:
But from this code it's obviously visible that that we want to order it.

Reply:
It is obvious from the point when `OrderedDict` appears.

Reply:
Done